bin/coach

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env ruby
+
+require "slop"
+require "coach/cli/provider_finder"
+
+begin
+  require File.join(Dir.pwd, "config/environment")
+rescue LoadError
+  puts <<~EOS
+    Could not load your Rails app
+    =============================
+
+    Currently the coach CLI assumes you have a config/environment.rb file that
+    we can load. We believe this is true of Rails apps in general.
+
+    Please raise an issue if that's not the case!
+
+    https://github.com/gocardless/coach/issues
+  EOS
+  exit 1
+end
+
+Slop.parse do
+  command "find-provider" do
+    run do |_, args|
+      unless args.length == 2
+        raise "Two args please!"
+      end
+      middleware_name = args[0]
+      value_name = args[1]
+
+      result = Coach::Cli::ProviderFinder.new(args[0], args[1]).find_provider
+
+      puts "Value `#{value_name}` is provided to `#{middleware_name}` by:\n\n"
+      puts result.to_a.join("\n")
+    end
+  end
+
+  command "find-chain" do
+    run do |_, args|
+      unless args.length == 2
+        raise "Two args please!"
```

Comment:
I presume Slop handles this for us and this edge case is unlikely? This error could be clearer anyway... Maybe write this more idiomatically like:

```ruby
middleware_name, value_name = *args
raise ArgumentError, "middleware_name and value_name required" unless middleware_name && value_name
```

Reply:
From what I can see, Slop doesn't give you any help with positional args so we have to do this bit ourselves. I've swapped to your version as it's better.

I'd like to get rid of Slop in favour of something else in the future tbh. It was just a convenient way to get this started. Maybe a follow-up PR?